proximal/examples/test_deconv_sv_psf.py

```diff
@@ -0,0 +1,153 @@
+# Spatially Varying PSF Deconvolution Algorithm based on:
+# Flicker & Rigaut, 2005 https://doi.org/10.1364/JOSAA.22.000504
+
+import sys
+
+sys.path.append('../../')
+
+import proximal as px
+from proximal.lin_ops.black_box import LinOpFactory
+from scipy.signal import fftconvolve
+import numpy as np
+
+############################################################
+
+
+def planewise_conv(G, im_shape, n_psf: int):
+    ''' Planewise convolution.
+
+        For each PSF with small supports, simulate the image blur with 2D
+        shift-invariant convolution.
+    '''
+    assert im_shape[0] * im_shape[1] == G.shape[0]
+    psf_mode = G[:, :n_psf].T.reshape((n_psf, *im_shape))
+
+    def forward(im_src, im_sim):
+        assert np.all(im_src.shape == im_sim.shape[1:])
+        assert im_sim.shape[0] == n_psf
+
+        for i in range(n_psf):
+            im_sim[i, :] = fftconvolve(psf_mode[i], im_src, mode='same')
+
+    def adjoint(im_src, im_sim):
+        assert im_src.shape[0] == n_psf
+        assert np.all(im_src.shape[1:] == im_sim.shape)
+
+        im_sim[:] = 0
+        for i in range(n_psf):
+            im_sim[:] += fftconvolve(im_src[i],
+                                     np.rot90(psf_mode[i], k=2),
+                                     mode='same')
+
+    # Estimate the largest eigenvalue of convT[ conv(x)] .
+    scale_factor = np.sum(psf_mode, axis=(1, 2)).max() * n_psf
+    blur_op = LinOpFactory(im_shape, (n_psf, *im_shape),
+                           forward,
+                           adjoint,
+                           norm_bound=scale_factor)
```

Comment:
The `norm_bound` is the largest singular value of `K^T K`. But, since the weights in the examples contains negative values, I don't know whether analytical solution exists. Please revise the `norm_bound` estimate accordingly. Or, rely on ProxImaL itself to estimate the eigenvalue for you.